Support for live audit data (and misc changes)

djarotech opened the pull request:
In order to see live changes to the kube-apiserver-audit.log file, I mounted a volume with the File that is constantly being updates by Kubernetes, upon changes to a custom resource. Furthermore, to support live audit data, I made changes so that the project can be run on minikube. However, since minikube does not officially support audit logging, I am using a sample minikube audit log  that I have generated using a Postgres custom resource and applying a few functions to generate the log. Now, the project continuously rebuilds its provenance history when it is on Kubernetes but not minikube since auditing is not supported(it doesn'nt need torebuild). To check if I am on minikube or kubernetes I use status.hostIP in rc.yaml.

devdattakulkarni commented on README.md:
```diff
@@ -138,9 +139,23 @@ dep version //to verify that it is installed correctly
 git clone https://github.com/cloud-ark/kubeprovenance.git $GOPATH/src/github.com/cloud-ark<br/>
 cd $GOPATH/src/github.com/cloud-ark/kubeprovenance <br/>
+
+go get github.com/emicklei/go-restful<br/>
```
Curious as to why do we need to separately do 'go get' for these libraries?

Can we not include them in Gopgk.toml and have them installed by dep ensure?

djarotech replied:
Sure !

devdattakulkarni commented on pkg/provenance/provenance.go:
```diff
@@ -480,48 +505,47 @@ func (o ObjectLineage) FullDiff(vNumStart, vNumEnd int) string {
-					fmt.Fprintf(b, "Found diff on attribute %s:\n", fieldName)
-					fmt.Fprintf(b, "\tVersion %d: %s\n", vNumStart, strArray1)
-					fmt.Fprintf(b, "\tVersion %d: %s\n", vNumEnd, strArray2)
+func getDiff(b *strings.Builder, fieldName string, data1, data2 interface{}, vNumStart, vNumEnd int) string {
```
Strange that getDiff gets included again!!

Are there any real changes that you made? If so, what I meant earlier was -- can you not include those changes in this PR?

I would like this PR to only focus on live data related changes. The formatting changes (like adding extra spaces) are fine. But if other logic changes, such as bug fixes, are included in this PR, they will get hidden under wrong commit message. And that will be problematic if later we want to go back and see where the changes are being made.

If there are no real changes to getDiff but Github is still showing it then I guess we cannot do much.

danielpygo replied:
It looks like all the code is actually the same.
I think It is just showing here because the line numbers are different due to the code at the top I added for live data?

 The good news is that func all() is here in both, and the method is the same even though it marks it as a diff

djarotech replied:
yeah it is weird ! :(

No I did not make any real changes besides ~20 lines of code for the live data

I understand, I did not make any logic changes or bug fixes .

devdattakulkarni replied:
I see. In that case it is fine.

Update the PR with Gopkg.toml after you test it. Then I think it will be ready for merge.

djarotech replied:
Ok

djarotech committed:
Support for live audit data (and misc changes)

In order to see live changes to the kube-apiserver-audit.log file, I mounted a volume with the File that is constantly being updates by Kubernetes, upon changes to a custom resource. Furthermore, to support live audit data, I made changes so that the project can be run on minikube. However, since minikube does not officially support audit logging, I am using a sample minikube audit log  that I have generated using a Postgres custom resource and applying a few functions to generate the log. Now, the project continuously rebuilds its provenance history when it is on Kubernetes but not minikube since auditing is not supported(it doesn'nt need torebuild). To check if I am on minikube or kubernetes I use status.hostIP in rc.yaml.
